src/extension.ts

```diff
@@ -31,7 +31,7 @@ interface TerminalLinkWithData extends vscode.TerminalLink {
 
 const ExpandMacroTitle = 'Expand macro result'
 
-export let defaultClient: LanguageClient;
+export let defaultClient: LanguageClient | null;
```

Comment:
Is this sometimes null when there are no untitled files open?

Reply:
Yes, we create it on demand and it only ever happens when opening a single file. Then rootUri is null and workspace array is empty.
In case of single dir/multiple workspaces it's not going to be created at all

Reply:
Okay, makes sense